pkgs/applications/science/astronomy/openspace/default.nix

```diff
@@ -0,0 +1,78 @@
+{ stdenv, fetchFromGitHub, makeWrapper, cmake
+, curl, boost, gdal, glew, soil
+, libX11, libXi, libXxf86vm, libXcursor, libXrandr, libXinerama }:
+
+stdenv.mkDerivation rec {
+  version = "0.11.1";
+  name    = "openspace-${version}";
+  
+  src = fetchFromGitHub {
+    owner  = "OpenSpace";
+    repo   = "OpenSpace";
+    rev    = "a65eea61a1b8807ce3d69e9925e75f8e3dfb085d";
+    sha256 = "0msqixf30r0d41xmfmzkdfw6w9jkx2ph5clq8xiwrg1jc3z9q7nv";
+    fetchSubmodules = true;
+  };
+  
+  buildInputs = [
+    makeWrapper cmake
+    curl boost gdal glew soil
+    libX11 libXi libXxf86vm libXcursor libXrandr libXinerama
+  ];
+
+  patches = [
```

Comment:
Please use fetchpatch for all patches available upstream.

Reply:
The only patch available upstream is `glm.patch`. I could fetch the corresponding commit patch from glm's github repo, but glm is a submodule of openspace, so I can't apply the patch to the openspace source tree. Also, I made this patch by picking a whole file from a specific commit and replacing the several matching files in the source tree, which might not lead to the same result as applying the commit patch.

Reply:
Maybe I could fetch this file from github with `fetchurl`, and manually replace the matching files in the source tree ?

Reply:
`fetchpatch` takes an optional argument `extraPrefix`. Are you sure this is not good enough?

Reply:
thanks for the tip, I will try

Reply:
Done with `fetchurl`. I tried the `fetchpatch` method but it did not succeed because the commit patch was not applicable (the version of the file before the commit and the version of the source tree one did not match).